src/ast.js

```diff
@@ -3,6 +3,7 @@ const {symbolMap} = require('./constants');
 
 function traversSourceCode(source, parse) {
     const root = AST.parse(source, parse);
+    const getSourceByLines = sourceGetter(source);
```

Comment:
You can calculate the lines once in `traversSourceCode`
```
     const sourceSplitLines = source.split('\n');
```
And create function named `getLinesContent` which receives the lines and the location object, and returns the actual content.

Also `getLinesFromSource` could be simplified using `Array.prototype.slice` and `Array.prototype.join` functions, 
since you only need to get certain lines from the lines array and join them with new lines ('\n')